### DISCUSSION TITLE
Emacs style incremental search?

### DISCUSSION THREAD

**Discussion by perapp** [helix-editor/helix#9672] [Ideas] [answered]
Hi!
I've been using Emacs as my editor for the last 25 years and thought I would finally learn vi or vim. But found helix and I think I might actually learn to use it after 25y of Emacs hacking 😱 . One feature of Emacs I use a lot but haven't found yet is the incremental search, C-s in Emacs. Some more detail here: [emacs doc](https://www.gnu.org/software/emacs/manual/html_node/emacs/Incremental-Search.html),  [stackoverflow](https://stackoverflow.com/questions/40192919/how-to-do-incremenatal-search-in-vim-like-it-is-done-in-emacs). Very convenient when you don't know how much of a search term you need to search for in order to find the correct spot.

Is there a similar search in helix?

**Comment by the-mikedavis** [comment, marked as answer]
We jump to the current match like Emacs' incremental search in `/`. Are you looking for the ability to jump between matches while searching (i.e. https://www.gnu.org/software/emacs/manual/html_node/emacs/Repeat-Isearch.html `isearch-repeat-forward` / `isearch-repeat-backward`)?

**Reply by perapp** [reply]
Hmm. Sorry. Not sure what I was doing. Probably got my Helix and Vim sessions mixed up. Yes, in Helix the `/` search works exactly as I would want! I should stick with Helix and forget about Vim for now. Thanks @the-mikedavis. Will keep using Helix every now and then. It's one of the few editors over the years I've felt can become my new day to day editor in the future.
